nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -157,18 +172,34 @@ def loss_func(output_tensor):
                     vocab_parallel_logits=output_tensor, target=labels, higher_stability=True
                 )
 
-                loss, acc_chosen = self.loss_func(
-                    per_token_logps, ref_logprobs, labels[:, 1:], average_log_probs=self.avg_log_probs
+                preference_loss, acc_chosen = self.loss_func(
+                    per_token_logps, ref_logprobs, labels[:, 1:], gt_rewards, average_log_probs=self.avg_log_probs
                 )
 
-                reduced_loss = average_losses_across_data_parallel_group([loss])
-                reduced_acc = average_losses_across_data_parallel_group([acc_chosen])
+                sft_loss = torch.zeros_like(preference_loss)
+                if self.sft_loss_weight != 0:
+                    sft_loss = self.sft_loss_func(per_token_logps, labels[:, 1:], average_log_probs=self.avg_log_probs)
+                loss = self.preference_loss_weight * preference_loss + self.sft_loss_weight * sft_loss
```

Comment:
Have you done any testing on this logic, does it perhaps work better when the weights are tied somehow? Could you maybe test something like: `self.preference_loss_weight * preference_loss + (1.0 - self.preference_loss_weight) * sft_loss`

This would also serve to eliminate another parameter that needs to be tuned by hand via grid search. These kinds of unbounded hyperparameters are not great, because it's difficult to tune them and they have no meaningful boundedness or numeraire.

Reply:
I believe @Kipok wanted this to be flexible in case they need to tune it, but maybe in the config we can do what you said?

Reply:
Yeah, since we don't have any experimental results yet, it's best to make it flexible and when we verify that a particular combination is the most effective, we can potentially set it as default or even remove one of the parameters

Reply:
Hi, 

Can we have an option to control for average_log_probs for both the dpo and the SFT loss?
I have a PR in my forked repo which does that - https://github.com/NVIDIA/NeMo-Aligner/compare/main...shtoshni:NeMo-Aligner:avg_log_prob

Thanks,
Shubham

Reply:
just for my understanding why would we need to set avg log prob = False in SFT? each sample will contribute differently to the loss that way

Reply:
Yeah, agreed that loss averaging for SFT is a reasonable choice for most cases I can imagine. Right now the avg_log_probs param can change that. So maybe let's keep averaging true for SFT by default.

Reply:
yeah sounds good. 8feb093655d753bdf63a487bbe59f84d8a1079d2 should fix it. I ran the CI last night and it's queued atm to check this

Reply:
In my opinion, the avg_log_probs should mostly be False in SFT. Looking at the SFT loss, it is 1/n sum_{i=1}^n log p(y_i | x_i), without penalizing each example based on its length.